src/Application/Commands/ArchiveShoppingList/Validation/Rules/ShoppingListSlugDoesNotExist.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Lindyhopchris\ShoppingList\Application\Commands\ArchiveShoppingList\Validation\Rules;
+
+use Lindyhopchris\ShoppingList\Application\Commands\ArchiveShoppingList\ArchiveShoppingListModel;
+use Lindyhopchris\ShoppingList\Application\Commands\ArchiveShoppingList\Validation\ArchiveShoppingListRuleInterface;
+use Lindyhopchris\ShoppingList\Persistance\ShoppingListRepositoryInterface;
+use Lindyhopchris\ShoppingList\Common\Validation\ValidationMessageStack;
+
+class ShoppingListSlugDoesNotExist implements ArchiveShoppingListRuleInterface
+{
+    /**
+     * @var ShoppingListRepositoryInterface
+     */
+    private ShoppingListRepositoryInterface $repository;
+
+    /**
+     * @param ShoppingListRepositoryInterface $repository
+     */
+    public function __construct(ShoppingListRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+    public function validate(ArchiveShoppingListModel $model): ValidationMessageStack
```

Comment:
Required: there needs to be a blank line between the `__construct` and `validate` methods. Also, the `validate` method needs a docblock.